aptos-move/framework/aptos-framework/sources/IterableTable.move

```diff
@@ -0,0 +1,184 @@
+module AptosFramework::IterableTable {
```

Comment:
Should we also plan to implement some sort of `FatTable` where a `Table` is split into multiple buckets and within each bucket, elements can be stored in a BigVector? This will reduce the slots we use to store tables in storage at the cost of operations on the table becoming expensive in case of hash collisions?

Reply:
yes, I was planning to implement that too, although I'm worried that hash(k) in move is gonna be expensive

Reply:
when is `hash` called?

Reply:
what I imagine for the `FatTable` is a vector based hash map implementation, we need to hash the key into slots and use open addressing or something to deal with collisions